tests/unit/s2n_server_hello_retry_test.c

```diff
@@ -250,19 +240,17 @@ int main(int argc, char **argv)
         /* Setup the client to receive a HelloRetryRequest */
         memcpy_check(client_conn->secure.server_random, hello_retry_request_random_test_buffer, S2N_TLS_RANDOM_DATA_LEN);
         client_conn->server_protocol_version = S2N_TLS13;
+        client_conn->client_protocol_version = S2N_TLS13;
 
         /* Setup the handshake type and message number to simulate a condition where a HelloRetry should be sent */
-        client_conn->handshake.handshake_type = NEGOTIATED | FULL_HANDSHAKE;
+        client_conn->handshake.handshake_type = NEGOTIATED | HELLO_RETRY_REQUEST | FULL_HANDSHAKE;
         EXPECT_SUCCESS(s2n_set_hello_retry_required(client_conn));
         client_conn->handshake.message_number = 1;
 
         /* Parse the key share */
         EXPECT_SUCCESS(s2n_extensions_server_key_share_recv(client_conn, extension_stuffer));
         EXPECT_EQUAL(s2n_stuffer_data_available(extension_stuffer), 0);
 
-        /* This curve will be NULL even though the extension succeeded */
```

Comment:
Yes, this will no longer be NULL because `client_conn->secure.server_ecc_evp_params` holds the value of the server negotiated curve for which the client will be generating a keyshare in ClientHello2.

Reply:
Can we verify that then?

Reply:
Yep, on it.